k9mail/src/main/java/com/fsck/k9/message/extractors/MessageFulltextCreator.java

```diff
@@ -0,0 +1,59 @@
+package com.fsck.k9.message.extractors;
+
+
+import android.support.annotation.NonNull;
+
+import com.fsck.k9.helper.HtmlConverter;
+import com.fsck.k9.mail.Message;
+import com.fsck.k9.mail.Part;
+import com.fsck.k9.mail.internet.MessageExtractor;
+import com.fsck.k9.mail.internet.MimeUtility;
+
+
+public class MessageFulltextCreator {
+    private static final int MAX_CHARACTERS_CHECKED_FOR_FTS = 200*1024;
+
+
+    private final TextPartFinder textPartFinder;
+    private final EncryptionDetector encryptionDetector;
+
+
+    MessageFulltextCreator(TextPartFinder textPartFinder, EncryptionDetector encryptionDetector) {
+        this.textPartFinder = textPartFinder;
+        this.encryptionDetector = encryptionDetector;
+    }
+
+    public static MessageFulltextCreator newInstance() {
+        TextPartFinder textPartFinder = new TextPartFinder();
+        EncryptionDetector encryptionDetector = new EncryptionDetector(textPartFinder);
+        return new MessageFulltextCreator(textPartFinder, encryptionDetector);
+    }
+
+    public String createFulltext(@NonNull Message message) {
+        if (encryptionDetector.isEncrypted(message)) {
+            return null;
+        }
+
+        return extractText(message);
+    }
+
+    private String extractText(Message message) {
+        Part textPart = textPartFinder.findFirstTextPart(message);
```

Comment:
This method is called three times during message creation: for preview, for encryption-detection, and for fulltext extraction. Might be worth diluting the separation of concerns a little for performance?